Makefile

```diff
@@ -0,0 +1,22 @@
+DOCKER := $(shell { command -v podman || command -v docker; })
+
+.PHONY: all clean distclean
+
+all:
+	mkdir -p firmware
+	$(DOCKER) build --tag zmk --build-arg USER_ID=$(shell id -u) .
+	$(DOCKER) volume create build
+	$(DOCKER) run --rm -it --name zmk \
+		-v build:/app \
+		-v $(PWD)/build.yaml:/app/build.yaml:ro \
+		-v $(PWD)/config:/app/config:ro \
+		-v $(PWD)/firmware:/app/firmware \
```

Comment:
It might be nice to use volume mounts for the few west module directories that we currently *cache* as part of our GHA workflow, so that the west update portion doesn't have to pull *everything* every single time you run this Makefile.

Reply:
So I believe the `west update` portion already is cached in the container. I've attached the output from a cached build and I don't see anything related to this:

[make.txt](https://github.com/zmkfirmware/unified-zmk-config-template/files/12795404/make.txt)

Reply:
Actually, I just use one volume for the `/app` folder